src/segments/git.go

```diff
@@ -538,6 +541,59 @@ func (g *Git) setBranchStatus() {
 	g.BranchStatus = getBranchStatus()
 }
 
+func (g *Git) setPushStatus() {
```

Comment:
I would argue we need to add a separate property to enable fetching this. Otherwise everyone will wait for this to complete. We can add `fetch_push_status` to avoid calling this.